KBH.ASM

```diff
@@ -0,0 +1,101 @@
+IDEAL
+P386
+MODEL FLAT, C
+ASSUME cs:_TEXT,ds:FLAT,es:FLAT,fs:FLAT,gs:FLAT
+
+include "kbh.inc"
+
+CODESEG
+proc install_keyboard_handler ;; might wanna add ebp and esp
+    uses eax, ebx, ecx, edx, edi, esi
+
+
+
+    mov ecx, (128 / 2) + 1
+    mov edi, offset __keyb_keyboardState
+    xor eax, eax
+    rep stosw ;; we fill the keyboard state with 0
+
+    ;; saving old handler data
+    push es
+    mov eax, 3509h ; get current interrupt handler 09h
```

Comment:
This is not very readable. Maybe split it up in 
`mov al, 35h ; get interrupt handler
mov ah, 09h ; interrupt handler 9`